lib/private/Files/ObjectStore/ObjectStoreStorage.php

```diff
@@ -303,13 +303,23 @@ public function fopen($path, $mode) {
 			case 'rb':
 				$stat = $this->stat($path);
 				if (is_array($stat)) {
+					$filesize = $stat['size'] ?? 0;
 					// Reading 0 sized files is a waste of time
-					if (isset($stat['size']) && $stat['size'] === 0) {
+					if ($filesize === 0) {
 						return fopen('php://memory', $mode);
 					}
 
 					try {
-						return $this->objectStore->readObject($this->getURN($stat['fileid']));
+						$handle = $this->objectStore->readObject($this->getURN($stat['fileid']));
+						if ($handle === false) {
+							return false; // keep backward compatibility
+						}
+						$streamStat = fstat($handle);
+						$actualSize = (int)$streamStat['size'] ?? -1;
```

Comment:
```suggestion
						$actualSize = $streamStat['size'] ?? -1;
```
Cast is not needed, there is the null coalesce operator for that.